{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "869f1f66-381c-48f3-9242-e22ca2ad8a56",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Biobase’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "\n",
      "\n",
      "Attaching package: ‘DelayedArray’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    aperm, apply, rowsum\n",
      "\n",
      "\n",
      "── \u001b[1mAttaching packages\u001b[22m ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.4.0     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.3     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.3     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m   masks \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiobase\u001b[39m::combine(), \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcount()\u001b[39m      masks \u001b[34mmatrixStats\u001b[39m::count()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m       masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32msimplify()\u001b[39m   masks \u001b[34mDelayedArray\u001b[39m::simplify()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m      masks \u001b[34mIRanges\u001b[39m::slice()\n",
      "\n",
      "here() starts at /projects/pfenninggroup/singleCell/Macaque_SealDorsalHorn_snATAC-seq\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "DataFrame with 6 rows and 16 columns\n",
       "           Barcode Unnamed..0\n",
       "          <factor>  <integer>\n",
       "1 AAACCCAAGGATACGC          1\n",
       "2 AAACCCAAGTAGGTTA          2\n",
       "3 AAACCCAGTCGTATTG          3\n",
       "4 AAACGAAAGGTGATAT          4\n",
       "5 AAACGAAGTACCAATC          5\n",
       "6 AAACGAAGTATCGAGG          6\n",
       "                                                                                              Sample\n",
       "                                                                                            <factor>\n",
       "1 /projects/pfenninggroup/machineLearningForComputationalBiology/SingleCell/dorsal_horn/P0/Solo.out/\n",
       "2 /projects/pfenninggroup/machineLearningForComputationalBiology/SingleCell/dorsal_horn/P0/Solo.out/\n",
       "3 /projects/pfenninggroup/machineLearningForComputationalBiology/SingleCell/dorsal_horn/P0/Solo.out/\n",
       "4 /projects/pfenninggroup/machineLearningForComputationalBiology/SingleCell/dorsal_horn/P0/Solo.out/\n",
       "5 /projects/pfenninggroup/machineLearningForComputationalBiology/SingleCell/dorsal_horn/P0/Solo.out/\n",
       "6 /projects/pfenninggroup/machineLearningForComputationalBiology/SingleCell/dorsal_horn/P0/Solo.out/\n",
       "        cxds_score         bcds_score      hybrid_score precluster      nUMI\n",
       "         <numeric>          <numeric>         <numeric>  <integer> <integer>\n",
       "1 151402.545658441  0.102374665439129 0.258837760826166          8     16448\n",
       "2 207183.175964405 0.0779373124241829  0.30332646065921          2     25551\n",
       "3  166627.21093315 0.0413215458393097 0.211647428100564          4      6860\n",
       "4 126728.563341809 0.0457363836467266 0.165571890834032         11     19303\n",
       "5 345115.402180292  0.333405524492264 0.757984117760983          9     21247\n",
       "6 82213.8421314786 0.0413389354944229 0.103994032699817          0      5786\n",
       "      nGene sample_short           unique_id precluster_sample      size_factor\n",
       "  <integer>     <factor>         <character>          <factor>        <numeric>\n",
       "1      5536           P0 AAACCCAAGGATACGC_P0              8_P0 3.71738116411669\n",
       "2      7119           P0 AAACCCAAGTAGGTTA_P0              2_P0 5.91423165132553\n",
       "3      3124           P0 AAACCCAGTCGTATTG_P0              4_P0   1.446859925597\n",
       "4      6062           P0 AAACGAAAGGTGATAT_P0             11_P0 4.29774412673016\n",
       "5      6543           P0 AAACGAAGTACCAATC_P0              9_P0 5.00222942830032\n",
       "6      2457           P0 AAACGAAGTATCGAGG_P0              0_P0  1.0823132527138\n",
       "    leiden   cell_type    ident\n",
       "  <factor>    <factor> <factor>\n",
       "1        6       GLUT6  AnnData\n",
       "2        4       GABA1  AnnData\n",
       "3        3 Astrocyte 1  AnnData\n",
       "4       12       GABA5  AnnData\n",
       "5        6       GLUT1  AnnData\n",
       "6        0     Oligo 1  AnnData"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get \n",
    "library(SingleCellExperiment)\n",
    "library(tidyverse)\n",
    "library(here)\n",
    "library(scater)\n",
    "library(scran)\n",
    "\n",
    "sce_rm = readRDS(here('data/tidy_data/rdas/full_integrated_macaque_dorsal_hornv3.sce.rds'))\n",
    "head(colData(sce_rm))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "380ff45b-e9d2-4f69-8e12-7967bda8fc87",
   "metadata": {},
   "outputs": [],
   "source": [
    "try_markers <- scran::findMarkers(\n",
    "  sce_rm, groups = sce_rm$cell_type, \n",
    "  pval.type = \"some\", direction = \"up\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2074a0af-2715-41d1-89bf-0bde79bb303e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>GLUT6</li><li>GABA1</li><li>Astrocyte 1</li><li>GABA5</li><li>GLUT1</li><li>Oligo 1</li><li>midVen</li><li>Oligo 2</li><li>Meninges</li><li>GABA4_1</li><li>Astrocyte 2</li><li>OPC</li><li>GLUT5</li><li>GABA4_2</li><li>Microglia</li><li>GLUT9</li><li>GABA2_1</li><li>GLUT4</li><li>GABA2_2</li><li>GLUT2</li><li>GABA3</li><li>GLUT3</li><li>GLUT10</li><li>GLUT11</li><li>GLUT7</li><li>GLUT8</li><li>Schwann</li><li>Ependymal cells</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'GLUT1'</li><li>'GLUT2'</li><li>'GLUT3'</li><li>'GLUT4'</li><li>'GLUT5'</li><li>'GLUT6'</li><li>'GLUT7'</li><li>'GLUT8'</li><li>'GLUT9'</li><li>'GLUT10'</li><li>'GLUT11'</li><li>'GABA1'</li><li>'GABA2_1'</li><li>'GABA2_2'</li><li>'GABA3'</li><li>'GABA4_1'</li><li>'GABA4_2'</li><li>'GABA5'</li><li>'midVen'</li><li>'Astrocyte 1'</li><li>'Astrocyte 2'</li><li>'Ependymal cells'</li><li>'Meninges'</li><li>'Microglia'</li><li>'Oligo 1'</li><li>'Oligo 2'</li><li>'OPC'</li><li>'Schwann'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item GLUT6\n",
       "\\item GABA1\n",
       "\\item Astrocyte 1\n",
       "\\item GABA5\n",
       "\\item GLUT1\n",
       "\\item Oligo 1\n",
       "\\item midVen\n",
       "\\item Oligo 2\n",
       "\\item Meninges\n",
       "\\item GABA4\\_1\n",
       "\\item Astrocyte 2\n",
       "\\item OPC\n",
       "\\item GLUT5\n",
       "\\item GABA4\\_2\n",
       "\\item Microglia\n",
       "\\item GLUT9\n",
       "\\item GABA2\\_1\n",
       "\\item GLUT4\n",
       "\\item GABA2\\_2\n",
       "\\item GLUT2\n",
       "\\item GABA3\n",
       "\\item GLUT3\n",
       "\\item GLUT10\n",
       "\\item GLUT11\n",
       "\\item GLUT7\n",
       "\\item GLUT8\n",
       "\\item Schwann\n",
       "\\item Ependymal cells\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'GLUT1'\n",
       "\\item 'GLUT2'\n",
       "\\item 'GLUT3'\n",
       "\\item 'GLUT4'\n",
       "\\item 'GLUT5'\n",
       "\\item 'GLUT6'\n",
       "\\item 'GLUT7'\n",
       "\\item 'GLUT8'\n",
       "\\item 'GLUT9'\n",
       "\\item 'GLUT10'\n",
       "\\item 'GLUT11'\n",
       "\\item 'GABA1'\n",
       "\\item 'GABA2\\_1'\n",
       "\\item 'GABA2\\_2'\n",
       "\\item 'GABA3'\n",
       "\\item 'GABA4\\_1'\n",
       "\\item 'GABA4\\_2'\n",
       "\\item 'GABA5'\n",
       "\\item 'midVen'\n",
       "\\item 'Astrocyte 1'\n",
       "\\item 'Astrocyte 2'\n",
       "\\item 'Ependymal cells'\n",
       "\\item 'Meninges'\n",
       "\\item 'Microglia'\n",
       "\\item 'Oligo 1'\n",
       "\\item 'Oligo 2'\n",
       "\\item 'OPC'\n",
       "\\item 'Schwann'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. GLUT6\n",
       "2. GABA1\n",
       "3. Astrocyte 1\n",
       "4. GABA5\n",
       "5. GLUT1\n",
       "6. Oligo 1\n",
       "7. midVen\n",
       "8. Oligo 2\n",
       "9. Meninges\n",
       "10. GABA4_1\n",
       "11. Astrocyte 2\n",
       "12. OPC\n",
       "13. GLUT5\n",
       "14. GABA4_2\n",
       "15. Microglia\n",
       "16. GLUT9\n",
       "17. GABA2_1\n",
       "18. GLUT4\n",
       "19. GABA2_2\n",
       "20. GLUT2\n",
       "21. GABA3\n",
       "22. GLUT3\n",
       "23. GLUT10\n",
       "24. GLUT11\n",
       "25. GLUT7\n",
       "26. GLUT8\n",
       "27. Schwann\n",
       "28. Ependymal cells\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'GLUT1'\n",
       "2. 'GLUT2'\n",
       "3. 'GLUT3'\n",
       "4. 'GLUT4'\n",
       "5. 'GLUT5'\n",
       "6. 'GLUT6'\n",
       "7. 'GLUT7'\n",
       "8. 'GLUT8'\n",
       "9. 'GLUT9'\n",
       "10. 'GLUT10'\n",
       "11. 'GLUT11'\n",
       "12. 'GABA1'\n",
       "13. 'GABA2_1'\n",
       "14. 'GABA2_2'\n",
       "15. 'GABA3'\n",
       "16. 'GABA4_1'\n",
       "17. 'GABA4_2'\n",
       "18. 'GABA5'\n",
       "19. 'midVen'\n",
       "20. 'Astrocyte 1'\n",
       "21. 'Astrocyte 2'\n",
       "22. 'Ependymal cells'\n",
       "23. 'Meninges'\n",
       "24. 'Microglia'\n",
       "25. 'Oligo 1'\n",
       "26. 'Oligo 2'\n",
       "27. 'OPC'\n",
       "28. 'Schwann'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] GLUT6           GABA1           Astrocyte 1     GABA5          \n",
       " [5] GLUT1           Oligo 1         midVen          Oligo 2        \n",
       " [9] Meninges        GABA4_1         Astrocyte 2     OPC            \n",
       "[13] GLUT5           GABA4_2         Microglia       GLUT9          \n",
       "[17] GABA2_1         GLUT4           GABA2_2         GLUT2          \n",
       "[21] GABA3           GLUT3           GLUT10          GLUT11         \n",
       "[25] GLUT7           GLUT8           Schwann         Ependymal cells\n",
       "28 Levels: GLUT1 GLUT2 GLUT3 GLUT4 GLUT5 GLUT6 GLUT7 GLUT8 GLUT9 ... Schwann"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sce_rm$cell_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "789d8a0b-8047-4014-a21a-8b0a4b27b8c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "class: SingleCellExperiment \n",
       "dim: 54894 12243 \n",
       "metadata(0):\n",
       "assays(2): counts logcounts\n",
       "rownames(54894): WASH7P MIR6859-1 ... DSTNP1 RPL23AP4\n",
       "rowData names(0):\n",
       "colnames(12243): 1 2 ... 12242 12243\n",
       "colData names(16): Barcode Unnamed..0 ... cell_type ident\n",
       "reducedDimNames(2): SCANORAMA UMAP\n",
       "spikeNames(0):\n",
       "altExpNames(0):"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sce_rm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a6d678d4-0fe8-4a95-9298-ca53fe422933",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "33005"
      ],
      "text/latex": [
       "33005"
      ],
      "text/markdown": [
       "33005"
      ],
      "text/plain": [
       "[1] 33005"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get genes that have non-zero expression (logcounts > 0)\n",
    "genes_any_expression = rownames(sce_rm[rowSums(logcounts(sce_rm)) > 0,] )\n",
    "length(genes_any_expression)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "2f022897-a724-41b1-b5fd-0cb0a45f186c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 54894\n"
     ]
    }
   ],
   "source": [
    "print(length(rownames(sce_rm)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2b8ed20e-1d75-4d6b-9014-1d8f8864f3fd",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>biomart</th><th scope=col>version</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSEMBL_MART_ENSEMBL</td><td>Ensembl Genes 109     </td></tr>\n",
       "\t<tr><td>ENSEMBL_MART_MOUSE  </td><td>Mouse strains 109     </td></tr>\n",
       "\t<tr><td>ENSEMBL_MART_SNP    </td><td>Ensembl Variation 109 </td></tr>\n",
       "\t<tr><td>ENSEMBL_MART_FUNCGEN</td><td>Ensembl Regulation 109</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 2\n",
       "\\begin{tabular}{ll}\n",
       " biomart & version\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSEMBL\\_MART\\_ENSEMBL & Ensembl Genes 109     \\\\\n",
       "\t ENSEMBL\\_MART\\_MOUSE   & Mouse strains 109     \\\\\n",
       "\t ENSEMBL\\_MART\\_SNP     & Ensembl Variation 109 \\\\\n",
       "\t ENSEMBL\\_MART\\_FUNCGEN & Ensembl Regulation 109\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 2\n",
       "\n",
       "| biomart &lt;chr&gt; | version &lt;chr&gt; |\n",
       "|---|---|\n",
       "| ENSEMBL_MART_ENSEMBL | Ensembl Genes 109      |\n",
       "| ENSEMBL_MART_MOUSE   | Mouse strains 109      |\n",
       "| ENSEMBL_MART_SNP     | Ensembl Variation 109  |\n",
       "| ENSEMBL_MART_FUNCGEN | Ensembl Regulation 109 |\n",
       "\n"
      ],
      "text/plain": [
       "  biomart              version               \n",
       "1 ENSEMBL_MART_ENSEMBL Ensembl Genes 109     \n",
       "2 ENSEMBL_MART_MOUSE   Mouse strains 109     \n",
       "3 ENSEMBL_MART_SNP     Ensembl Variation 109 \n",
       "4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 109"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ensembl site unresponsive, trying useast mirror\n",
      "\n",
      "Ensembl site unresponsive, trying useast mirror\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in getBM(attributes = c(\"external_gene_name\", \"ensembl_gene_id\"), : object 'genes' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in getBM(attributes = c(\"external_gene_name\", \"ensembl_gene_id\"), : object 'genes' not found\nTraceback:\n",
      "1. getBM(attributes = c(\"external_gene_name\", \"ensembl_gene_id\"), \n .     filters = \"external_gene_name\", values = genes, mart = ensembl)"
     ]
    }
   ],
   "source": [
    "# library(\"biomaRt\") \n",
    "# listMarts()\n",
    "# ensembl <- useMart(\"ensembl\")\n",
    "# datasets <- listDatasets(ensembl)\n",
    "# ensembl = useDataset(\"hsapiens_gene_ensembl\",mart=ensembl)\n",
    "# #ensembl = useDataset(\"mmulatta_gene_ensembl\", mart=ensembl)\n",
    "# options(max.print=1000000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1c68b3a6-c35b-4cdd-8ef7-a906f3d77549",
   "metadata": {},
   "outputs": [],
   "source": [
    "# y = getBM(attributes=c('external_gene_name','ensembl_gene_id'), filters = 'external_gene_name', values = genes, mart = ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a757a787-a527-45d7-ade7-f576885cee25",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>biomart</th><th scope=col>version</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSEMBL_MART_ENSEMBL</td><td>Ensembl Genes 109     </td></tr>\n",
       "\t<tr><td>ENSEMBL_MART_MOUSE  </td><td>Mouse strains 109     </td></tr>\n",
       "\t<tr><td>ENSEMBL_MART_SNP    </td><td>Ensembl Variation 109 </td></tr>\n",
       "\t<tr><td>ENSEMBL_MART_FUNCGEN</td><td>Ensembl Regulation 109</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 2\n",
       "\\begin{tabular}{ll}\n",
       " biomart & version\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSEMBL\\_MART\\_ENSEMBL & Ensembl Genes 109     \\\\\n",
       "\t ENSEMBL\\_MART\\_MOUSE   & Mouse strains 109     \\\\\n",
       "\t ENSEMBL\\_MART\\_SNP     & Ensembl Variation 109 \\\\\n",
       "\t ENSEMBL\\_MART\\_FUNCGEN & Ensembl Regulation 109\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 2\n",
       "\n",
       "| biomart &lt;chr&gt; | version &lt;chr&gt; |\n",
       "|---|---|\n",
       "| ENSEMBL_MART_ENSEMBL | Ensembl Genes 109      |\n",
       "| ENSEMBL_MART_MOUSE   | Mouse strains 109      |\n",
       "| ENSEMBL_MART_SNP     | Ensembl Variation 109  |\n",
       "| ENSEMBL_MART_FUNCGEN | Ensembl Regulation 109 |\n",
       "\n"
      ],
      "text/plain": [
       "  biomart              version               \n",
       "1 ENSEMBL_MART_ENSEMBL Ensembl Genes 109     \n",
       "2 ENSEMBL_MART_MOUSE   Mouse strains 109     \n",
       "3 ENSEMBL_MART_SNP     Ensembl Variation 109 \n",
       "4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 109"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ensembl site unresponsive, trying asia mirror\n",
      "\n",
      "Ensembl site unresponsive, trying useast mirror\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# library(\"biomaRt\") \n",
    "# listMarts()\n",
    "# ensembl <- useMart(\"ensembl\")\n",
    "# datasets <- listDatasets(ensembl)\n",
    "# ensembl = useDataset(\"hsapiens_gene_ensembl\",mart=ensembl)\n",
    "# #ensembl = useDataset(\"mmulatta_gene_ensembl\", mart=ensembl)\n",
    "# options(max.print=1000000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "eeee5115-8f17-4d78-a4b0-f0cc2df22a71",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"GLUT1\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT2\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT3\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT4\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT5\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT6\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT7\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT8\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT9\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT10\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT11\"\n",
      "[1] \"\"\n",
      "[1] \"GABA1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA2_1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA2_2\"\n",
      "[1] \"\"\n",
      "[1] \"GABA3\"\n",
      "[1] \"\"\n",
      "[1] \"GABA4_1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA4_2\"\n",
      "[1] \"\"\n",
      "[1] \"GABA5\"\n",
      "[1] \"\"\n",
      "[1] \"midVen\"\n",
      "[1] \"\"\n",
      "[1] \"Astrocyte 1\"\n",
      "[1] \"\"\n",
      "[1] \"Astrocyte 2\"\n",
      "[1] \"\"\n",
      "[1] \"Ependymal cells\"\n",
      "[1] \"\"\n",
      "[1] \"Meninges\"\n",
      "[1] \"\"\n",
      "[1] \"Microglia\"\n",
      "[1] \"\"\n",
      "[1] \"Oligo 1\"\n",
      "[1] \"\"\n",
      "[1] \"Oligo 2\"\n",
      "[1] \"\"\n",
      "[1] \"OPC\"\n",
      "[1] \"\"\n",
      "[1] \"Schwann\"\n",
      "[1] \"\"\n"
     ]
    }
   ],
   "source": [
    "# # with ensembl\n",
    "# for (ii in 1:length(try_markers) ){\n",
    "    \n",
    "#     celltype = names(try_markers[ii])\n",
    "#     x = try_markers[[ii]][try_markers[[ii]]$p.value < 0.05,]\n",
    "#     #genes = noquote(rownames(x))\n",
    "#     genes = rownames(x)\n",
    "    \n",
    "#     print(celltype)\n",
    "#     print('')\n",
    "    \n",
    "#     ### ensembl stuff\n",
    "#     ensembl_df = getBM(attributes=c('external_gene_name','ensembl_gene_id'), filters = 'external_gene_name', values = genes, mart = ensembl)\n",
    "#     ensembl_ids = noquote(ensembl_df$ensembl_gene_id)\n",
    "    \n",
    "#     filename = here( paste('data/tidy_data/celltype_specific_enhancers/tables/gene-based-ldsc/',celltype,'.csv', sep='') )\n",
    "\n",
    "#     #df <- data.frame(genes)\n",
    "#     df <- data.frame(ensembl_ids)\n",
    "#     #write.csv(df, filename, row.names=FALSE, col.names=FALSE)\n",
    "#     write.table(df, filename, row.names=FALSE, col.names=FALSE)\n",
    "    \n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2b6a0b09-2a68-42af-9cc3-5d316d92cbed",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"GLUT1\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT2\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT3\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT4\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT5\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT6\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT7\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT8\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT9\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT10\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT11\"\n",
      "[1] \"\"\n",
      "[1] \"GABA1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA2_1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA2_2\"\n",
      "[1] \"\"\n",
      "[1] \"GABA3\"\n",
      "[1] \"\"\n",
      "[1] \"GABA4_1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA4_2\"\n",
      "[1] \"\"\n",
      "[1] \"GABA5\"\n",
      "[1] \"\"\n",
      "[1] \"midVen\"\n",
      "[1] \"\"\n",
      "[1] \"Astrocyte1\"\n",
      "[1] \"\"\n",
      "[1] \"Astrocyte2\"\n",
      "[1] \"\"\n",
      "[1] \"Ependymal\"\n",
      "[1] \"\"\n",
      "[1] \"Meninges\"\n",
      "[1] \"\"\n",
      "[1] \"Microglia\"\n",
      "[1] \"\"\n",
      "[1] \"Oligo1\"\n",
      "[1] \"\"\n",
      "[1] \"Oligo2\"\n",
      "[1] \"\"\n",
      "[1] \"OPC\"\n",
      "[1] \"\"\n",
      "[1] \"Schwann\"\n",
      "[1] \"\"\n"
     ]
    }
   ],
   "source": [
    "# without ensembl\n",
    "for (ii in 1:length(try_markers) ){\n",
    "    \n",
    "    celltype = names(try_markers[ii])\n",
    "    x = try_markers[[ii]][try_markers[[ii]]$p.value < 0.05,]\n",
    "    genes = rownames(x)#noquote(rownames(x))\n",
    "    \n",
    "    celltype = gsub(\" \", \"\", celltype)\n",
    "    if (celltype == 'Ependymalcells'){\n",
    "        celltype = 'Ependymal'\n",
    "        }\n",
    "    print(celltype)\n",
    "    print('')\n",
    "    \n",
    "    filename = here( paste('data/tidy_data/celltype_specific_enhancers/tables/gene-based-ldsc/',celltype,'.csv', sep='') )\n",
    "\n",
    "    df <- data.frame(genes)\n",
    "    #write.csv(df, filename, row.names=FALSE, col.names=FALSE)\n",
    "    write.table(df, filename, row.names=FALSE, col.names=FALSE)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bc2d8c2d-79c4-4749-a7a7-af446c8f29fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"GLUT1\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT2\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT3\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT4\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT5\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT6\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT7\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT8\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT9\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT10\"\n",
      "[1] \"\"\n",
      "[1] \"GLUT11\"\n",
      "[1] \"\"\n",
      "[1] \"GABA1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA2_1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA2_2\"\n",
      "[1] \"\"\n",
      "[1] \"GABA3\"\n",
      "[1] \"\"\n",
      "[1] \"GABA4_1\"\n",
      "[1] \"\"\n",
      "[1] \"GABA4_2\"\n",
      "[1] \"\"\n",
      "[1] \"GABA5\"\n",
      "[1] \"\"\n",
      "[1] \"midVen\"\n",
      "[1] \"\"\n",
      "[1] \"Astrocyte1\"\n",
      "[1] \"\"\n",
      "[1] \"Astrocyte2\"\n",
      "[1] \"\"\n",
      "[1] \"Ependymalcells\"\n",
      "[1] \"\"\n",
      "[1] \"Meninges\"\n",
      "[1] \"\"\n",
      "[1] \"Microglia\"\n",
      "[1] \"\"\n",
      "[1] \"Oligo1\"\n",
      "[1] \"\"\n",
      "[1] \"Oligo2\"\n",
      "[1] \"\"\n",
      "[1] \"OPC\"\n",
      "[1] \"\"\n",
      "[1] \"Schwann\"\n",
      "[1] \"\"\n"
     ]
    }
   ],
   "source": [
    "# without ensembl\n",
    "for (ii in 1:length(try_markers) ){\n",
    "    \n",
    "    celltype = names(try_markers[ii])\n",
    "    x = try_markers[[ii]][try_markers[[ii]]$p.value < 0.05,]\n",
    "    genes = noquote(rownames(x))\n",
    "    \n",
    "    celltype = gsub(\" \", \"\", celltype)\n",
    "    if celltype == 'Ependymalcells'{\n",
    "        celltype = 'Ependymal'\n",
    "        }\n",
    "    print(celltype)\n",
    "    print('')\n",
    "    \n",
    "    #filename = here( paste('data/tidy_data/celltype_specific_enhancers/tables/gene-based-ldsc/',celltype,'.csv', sep='') )\n",
    "\n",
    "    #df <- data.frame(genes)\n",
    "    #write.csv(df, filename, row.names=FALSE, col.names=FALSE)\n",
    "    #write.table(df, filename, row.names=FALSE, col.names=FALSE)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9c546d5-d47f-420a-aa66-1af360f84b21",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
